samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/components/PlayKMLTourViewModel.kt

```diff
@@ -0,0 +1,158 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.playkmltour.components
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.mapping.ArcGISScene
+import com.arcgismaps.mapping.ArcGISTiledElevationSource
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Surface
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.kml.KmlContainer
+import com.arcgismaps.mapping.kml.KmlDataset
+import com.arcgismaps.mapping.kml.KmlNode
+import com.arcgismaps.mapping.kml.KmlTour
+import com.arcgismaps.mapping.kml.KmlTourController
+import com.arcgismaps.mapping.kml.KmlTourStatus
+import com.arcgismaps.mapping.layers.KmlLayer
+import com.arcgismaps.toolkit.geoviewcompose.SceneViewProxy
+import com.esri.arcgismaps.sample.playkmltour.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
+import java.io.File
+import kotlin.collections.forEach
+
+class PlayKMLTourViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val provisionPath: String by lazy { application.getExternalFilesDir(null)?.path.toString() +
+            File.separator +
+            application.getString(R.string.play_kml_tour_app_name)
+    }
+
+    // add elevation data
+    private val surface = Surface().apply {
+        elevationSources.add(ArcGISTiledElevationSource("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"))
+    }
+
+    // create a KML layer from a KML dataset with a KML tour
+    private val kmlDataSet = KmlDataset(provisionPath + File.separator + "Esri_tour.kmz")
+    private val kmlLayer = KmlLayer(kmlDataSet)
+
+    // create a scene with the surface and KML layer
+    val arcGISScene =
+        ArcGISScene(BasemapStyle.ArcGISImagery).apply {
+            baseSurface = surface
+            initialViewpoint = Viewpoint(39.8, -98.6, 10e7)
+            operationalLayers.add(kmlLayer)
+        }
+
+    val sceneViewProxy = SceneViewProxy()
+
+    private var kmlTour: KmlTour? = null
+    private val kmlTourController = KmlTourController()
+
+    private val _statusFlow: MutableStateFlow<KmlTourStatus> = MutableStateFlow(KmlTourStatus.NotInitialized)
+    val statusFlow = _statusFlow.asStateFlow()
```

Comment:
could these maybe more explictly named? This code gets read outside of an IDE on the website, so more verbose names can be helpful

_tourStatus or _tourStatusFlow maybe?